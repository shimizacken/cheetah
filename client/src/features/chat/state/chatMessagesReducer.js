import { PUBLISH_MESSAGE, EDIT_MESSAGE, DELETE_MESSAGE } from "./tests/chatMessagesConstants";
import produce from "immer";

const messagesInitialState = {};

export const messages = (state = messagesInitialState, action) => {
  if (action.type === PUBLISH_MESSAGE) {
    return {
      ...state,
      [action.message.id]: action.message
    };
  }

  if (action.type === EDIT_MESSAGE) {
    const nextState = produce(state, draftState => {
      draftState[action.message.id] = action.message;

      return draftState;
    });

    return nextState;
  }

  if (action.type === DELETE_MESSAGE) {
    const nextState = produce(state, draftState => {
      draftState[action.messageId].deleted = true;

      return draftState;
    });

    return nextState;
  }

  return state;
};
