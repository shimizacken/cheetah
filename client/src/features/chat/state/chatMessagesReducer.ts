import produce from 'immer';
import { AnyAction } from 'redux';
import {
  PUBLISH_MESSAGE,
  EDIT_MESSAGE,
  DELETE_MESSAGE,
  SET_MESSAGES
} from './chatMessagesConstants';

import type { ChatMessages } from '../../../state/store.types';

const MessagesInitialState: ChatMessages = {
  messages: {}
};

export const messages = (state = MessagesInitialState, action: AnyAction) => {
  if (action.type === SET_MESSAGES) {
    return action.messages;
  }

  if (action.type === PUBLISH_MESSAGE) {
    return {
      ...state,
      [action.message.id]: action.message
    };
  }

  if (action.type === EDIT_MESSAGE) {
    const nextState = produce(state, (draftState: ChatMessages) => {
      draftState.messages[action.message.id] = action.message;

      return draftState;
    });

    return nextState;
  }

  if (action.type === DELETE_MESSAGE) {
    const nextState = produce(state, (draftState) => {
      draftState.messages[action.messageId].deleted = true;

      return draftState;
    });

    return nextState;
  }

  return state;
};
