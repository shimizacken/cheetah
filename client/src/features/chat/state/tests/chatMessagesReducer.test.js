import { messages } from '../chatMessagesReducer';

describe('chatMessagesReducer test suite', () => {
    it('should return default state', () => {
        expect(messages()).toMatchSnapshot();
    });
});