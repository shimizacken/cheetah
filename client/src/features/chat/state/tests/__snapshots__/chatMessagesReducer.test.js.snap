// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chatMessagesReducer test suite should return default state 1`] = `Array []`;
